mscore/albummanager.h

```diff
@@ -17,35 +17,86 @@
 #include "abstractdialog.h"
 
 namespace Ms {
+class XmlWriter;
 class Movements;
+class MasterScore;
+class Score;
+class AlbumManagerDialog;
+class Album;
+struct AlbumItem;
+
+//---------------------------------------------------------
+//   AlbumManagerItem
+//---------------------------------------------------------
+
+struct AlbumManagerItem : public QObject {
+    Q_OBJECT
+
+public:
+    // can I convert these to references?
+    AlbumItem* albumItem;
+    QTableWidgetItem* listItem;
+    QTableWidgetItem* listDurationItem;
+
+    AlbumManagerItem(AlbumItem* albumItem, QTableWidgetItem* listItem, QTableWidgetItem* listDurationItem);
+    ~AlbumManagerItem();
+    void setEnabled(bool b);
+};
 
 //---------------------------------------------------------
 //   AlbumManager
 //---------------------------------------------------------
 
-class AlbumManager : public AbstractDialog, public Ui::AlbumManager
+class AlbumManager final : public QDockWidget, public Ui::AlbumManager
 {
     Q_OBJECT
-    Movements * album;
 
-    virtual void hideEvent(QHideEvent*);
+private:
+    AlbumManagerDialog* settingsDialog        { nullptr };
+    Album* _album                             { nullptr };
+    std::vector<AlbumManagerItem*> _items     {};
+    MasterScore* tempScore                    { nullptr };
+
+    int dragEnterIndex      { -1 };
+    int dropIndex           { -1 };
+    int playbackIndex       { -1 };
+    bool continuing         { false };
+
+    virtual void hideEvent(QHideEvent*) override;
+    void updateDurations();
 
 private slots:
-    void addClicked();
-    void addNewClicked();
-    void upClicked();
-    void downClicked();
-    void removeClicked();
-    void currentScoreChanged(int);
-    void itemChanged(QListWidgetItem*);     // score name in list is edited
-    void buttonBoxClicked(QAbstractButton*);
+    void addClicked(bool throwaway = false);
```

Comment:
You don't have to give these `throwaway` parameters a name. This is valid syntax:

```c++
void addClicked(bool = false);
```

Doing this avoids compiler warnings that appear when you give something a name and never actually use it.

See [this comment](https://github.com/musescore/MuseScore/pull/6145/files#r443144558) for an alterative solution using `Q_UNUSED(throwaway)` in the `.cpp` file.

Reply:
Addressed in the latest push.